apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet.IntegrationTests/ErrorReporting/ErrorReportingTest.cs

```diff
@@ -0,0 +1,271 @@
+
+            if (Utils.IsWindows())
+            {
+                Assert.False(string.IsNullOrWhiteSpace(errorEvent.Context.ReportLocation.FilePath));
+                Assert.True(errorEvent.Context.ReportLocation.LineNumber > 0);
+            }
+            Assert.Equal(functionName, errorEvent.Context.ReportLocation.FunctionName);
+        }
+
+        private class ErrorReportingTestApplication : HttpApplication
+        {
+            public void Configuration(IAppBuilder app)
+            {
+                HttpConfiguration config = new HttpConfiguration();
+                Register(config);
+                app.UseWebApi(config);
+            }
+
+            private static void Register(HttpConfiguration config)
+            {
+                config.Services.Add(typeof(System.Web.Http.ExceptionHandling.IExceptionLogger),
+                    ErrorReportingExceptionLogger.Create(ErrorReportingController.ProjectId, ErrorReportingController.Service, ErrorReportingController.Version));
+
+                config.IncludeErrorDetailPolicy = IncludeErrorDetailPolicy.Always;
+
+                // Web API routes
+                config.MapHttpAttributeRoutes();
+
+                config.Routes.MapHttpRoute(
+                    name: "DefaultApi",
+                    routeTemplate: "api/{controller}/{action}/{id}",
+                    defaults: new { id = RouteParameter.Optional }
+                );
+                config.Routes.MapHttpRoute(
+                    name: "MessageHandler",
+                    routeTemplate: "handler/{id}",
+                    null, null,
+                    new ThrowErrorHandler());
+            }
+        }
+    }
+
+    /// <summary>
+    /// A simple http handler that just throws an exception with the <see cref="_testId"/>
+    /// as the message.
+    /// </summary>
+    public class ThrowErrorHandler : HttpMessageHandler
+    {
+        protected override Task<HttpResponseMessage> SendAsync(
+            HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            string id = request.GetRouteData().Values["id"].ToString();
+            string message = ErrorReportingController.GetMessage(nameof(SendAsync), id);
+            throw new Exception(message);
+        }
+    };
+
+    /// <summary>
+    /// A controller for the <see cref="ErrorReportingTestApplication"/> used to log exceptions.
+    /// </summary>
+    public class ErrorReportingController : ApiController
+    {
+        public const string Service = "service-name";
+        public const string Version = "version-id";
+        public static readonly string ProjectId = Utils.GetProjectIdFromEnvironment();
+
+        private readonly IExceptionLogger _exceptionLogger;
+        public ErrorReportingController()
+        {
+            _exceptionLogger = GoogleExceptionLogger.Create(ProjectId, Service, Version);
+        }
+
+        /// <summary>Catches and handles a thrown <see cref="Exception"/>.</summary>
```

Comment:
suggestion: Throws and catches an exception

Reply:
Yep, done.